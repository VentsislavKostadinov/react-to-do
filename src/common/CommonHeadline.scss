@import '../style/font.scss';

p {
  color: white !important ; 
  font-family: "Poppins-Bold", sans-serif;
}