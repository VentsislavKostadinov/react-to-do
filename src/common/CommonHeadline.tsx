import { CommonHeadlineProps } from '../model/CommonHeadlineProps.types'
import classes from '../style/CommonHeadline.module.scss'

export const CommonHeadline = ({
    title,
    heading,
    dataTestid,
}: CommonHeadlineProps) => (
    <p data-testid={dataTestid} className={`${classes.title} ${heading}`}>
        {title}
    </p>
)
