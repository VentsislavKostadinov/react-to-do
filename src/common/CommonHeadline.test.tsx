import React from 'react'
import { CommonHeadline } from './CommonHeadline'
import { CommonHeadlineProps } from '../model/CommonHeadlineProps.types'
import { render, screen } from '@testing-library/react'

describe('CommonHeadline component', () => {
    const commonHeadlineProps: CommonHeadlineProps = {
        dataTestid: 'to-do-app',
        title: 'To Do App',
        heading: 'h2',
    }

    it('it renders correctly', () => {
        const { asFragment } = render(
            <CommonHeadline {...commonHeadlineProps} />,
        )
        expect(asFragment()).toMatchSnapshot()
    })

    it('it is visible and correct props', () => {
        render(<CommonHeadline {...commonHeadlineProps} />)

        const title = screen.getByTestId('to-do-app')
        expect(title).toHaveTextContent(commonHeadlineProps.title)
        expect(title).toBeInTheDocument
    })
})
