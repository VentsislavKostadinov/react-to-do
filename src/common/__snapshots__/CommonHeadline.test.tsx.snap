// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonHeadline component it renders correctly 1`] = `
<DocumentFragment>
  <p
    class="title h2"
    data-testid="to-do-app"
  >
    To Do App
  </p>
</DocumentFragment>
`;
